catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/BroadcastBricksTest.java

```diff
@@ -0,0 +1,143 @@
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.bricks.BroadcastBrick;
+import org.catrobat.catroid.content.bricks.BroadcastWaitBrick;
+import org.catrobat.catroid.test.utils.TestUtils;
+import org.catrobat.catroid.ui.SpriteActivity;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityInstrumentationRule;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static android.support.test.espresso.action.ViewActions.typeText;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.instanceOf;
+
+@RunWith(AndroidJUnit4.class)
+public class BroadcastBricksTest {
+
+	@Rule
+	public BaseActivityInstrumentationRule<SpriteActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(SpriteActivity.class, SpriteActivity.EXTRA_FRAGMENT_POSITION, SpriteActivity.FRAGMENT_SCRIPTS);
+
+	@Before
+	public void setUp() throws Exception {
+		createProject(this.getClass().getSimpleName());
+		baseActivityTestRule.launchActivity();
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		TestUtils.deleteProjects(this.getClass().getSimpleName());
+	}
+
+	@Test
+	public void testAllBroadcastBricksSpinnersShowTheNewAddedMessage() {
+		String message = "Oida!";
+		int firstBrickPosition = 0;
+
+		createNewMessageOnSpinner(firstBrickPosition, message);
+		onBrickAtPosition(firstBrickPosition)
+				.onSpinner(R.id.brick_broadcast_spinner)
+				.checkShowsText(message);
+
+		checkBricksSpinnerHasAndShowsText(1, message);
+		checkBricksSpinnerHasAndShowsText(2, message);
+		checkBricksSpinnerHasAndShowsText(3, message);
+		checkBricksSpinnerHasAndShowsText(4, message);
+		checkBricksSpinnerHasAndShowsText(5, message);
+	}
+
+	private void createNewMessageOnSpinner(int position, String message) {
+		onBrickAtPosition(position)
+				.onSpinner(R.id.brick_broadcast_spinner)
+				.performSelectNameable(R.string.new_option);
+
+		onView(allOf(withId(R.id.input_edit_text), isDisplayed(), instanceOf(EditText.class)))
+				.perform(typeText(message));
+		closeSoftKeyboard();
+
+		onView(withId(android.R.id.button1))
+				.perform(click());
+
+		onBrickAtPosition(position)
+				.onSpinner(R.id.brick_broadcast_spinner)
+				.checkShowsText(message);
+	}
+
+	private void checkBricksSpinnerHasAndShowsText(int brickPosition, String message) {
```

Comment:
is it necessary to select all messages? spinner handling in espresso is flaky by design and doing this will cause problems.

Reply:
not really. test modified